### PR TITLE
Added new tutorial tic-tac-toe for Seahorse

### PR DESCRIPTION
**Pull request by lostintime101**
Plan to upload several Seahorse tutorials. 
This being the first. 

Separate PR for the images coming

### CODE REVIEW

File: client/src/tutorials/Tic-tac-toe/files/client.ts.raw
Comment (acheroncrypto) [inline client/src/tutorials/Tic-tac-toe/files/client.ts.raw:1]: Could you remove this if it's not being used?

File: client/src/tutorials/Tic-tac-toe/files/index.test.ts.raw [hunk 1]
```diff
@@ -0,0 +1,69 @@
+
+describe("Tictactoe game", async () => {
+  const systemProgram = anchor.web3.SystemProgram;
+  const player1 = pg.wallet.publicKey;
+  const player2 = pg.wallet.publicKey;
```
Comment (acheroncrypto) [inline client/src/tutorials/Tic-tac-toe/files/index.test.ts.raw:5]: FYI, we can use multiple wallets at the same time here:

```
pg.wallets.<walletName>
```

(Add account from wallet settings)

File: client/src/tutorials/Tic-tac-toe/files/index.test.ts.raw [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    pg.program.programId
+  );
+
+  it("Create Game!", async () => {
+    let tx = await pg.program.methods
+      .initGame(player1, player2)
+      .accounts({
+        owner: pg.wallet.publicKey,
+        game: game,
+        systemProgram: systemProgram.programId,
```
Comment (acheroncrypto) [inline client/src/tutorials/Tic-tac-toe/files/index.test.ts.raw:24]: `systemProgram` can be omitted, up to you.

File: client/src/tutorials/Tic-tac-toe/files/index.test.ts.raw [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      .accounts({
+        player: player1,
+        gameData: game,
+      })
+      .rpc();
+
+    await pg.connection.confirmTransaction(tx);
+    console.log("Turn 1 signature", tx);
+
+    const game_account = await pg.program.account.game.fetch(game);
```
Comment (acheroncrypto) [inline client/src/tutorials/Tic-tac-toe/files/index.test.ts.raw:47]: Could you keep variable casing consistent with JS:

```suggestion
    const gameAccount = await pg.program.account.game.fetch(game);
```

File: client/src/tutorials/Tic-tac-toe/files/index.test.ts.raw [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      .accounts({
+        player: player2,
+        gameData: game,
+      })
+      .rpc();
+
+    await pg.connection.confirmTransaction(tx);
+    console.log("Turn 2 signature", tx);
+
+    const game_account = await pg.program.account.game.fetch(game);
```
Comment (acheroncrypto) [inline client/src/tutorials/Tic-tac-toe/files/index.test.ts.raw:66]: Same casing here:

```suggestion
    const gameAccount = await pg.program.account.game.fetch(game);
```

**Review by acheroncrypto — COMMENTED** [review]
Thanks, looking great!

One slight concern is the name of the tutorial since it's quite common e.g. Anchor also has a tutorial named [Tic-Tac-Toe](https://book.anchor-lang.com/anchor_in_depth/milestone_project_tic-tac-toe.html). It would be better to name it something like "Tic-Tac-Toe Seahorse" so that we can add this tutorial with other frameworks in the future. We also can't easily change tutorial names later because tutorial URLs are derived from tutorial names.

File: client/src/tutorials/Tic-tac-toe/files/index.test.ts.raw (hunk 1 above)
Comment (lostintime101) [reply client/src/tutorials/Tic-tac-toe/files/index.test.ts.raw:5]: Good to know. Keep the same for now i feel.

**Pull request closed without merging**